### DISCUSSION TITLE
Create new repository for stock synthesis automated workflows?

### DISCUSSION THREAD

**Discussion by k-doering-NOAA** [nmfs-stock-synthesis/stock-synthesis#430] [Team Posts]
Kelli and I were chatting today and think it may be good to have one place for all Stock Synthesis workflows (github actions). In the past we have discussed this, but it was not really possible to implement wthout [reusable workflows](https://docs.github.com/en/actions/learn-github-actions/reusing-workflows). 

We propose having a repo in the nmfs-stock-synthesis org, perhaps called "workflows" or "ss3-workflows", that contain workflows that can be shared across our R packages, and across the different repositories in nmfs-stock-synthesis.

Does anyone have thoughts about this?

**Comment by Rick-Methot-NOAA** [comment]
What is an example of such a workflow?  Is benefit/cost of such an effort clearly high?

**Comment by k-doering-NOAA** [comment]
Sure, here is a reusable workflow for checking R packages: https://github.com/k-doering-NOAA/reusable-workflows/blob/main/.github/workflows/r-cmd-check.yml

I set up SSMSE and r4ss to use it, which means less duplication: https://github.com/nmfs-fish-tools/SSMSE/blob/main/.github/workflows/call-r-cmd-check.yml and https://github.com/r4ss/r4ss/blob/main/.github/workflows/call-r-cmd-check.yml
ss3sim (and any R package) could also use the reusable workflow.

I think the benefit is it will be easier to update the workflows all at once and all the workflows will be in one place., in the same repo We can move the stock synthesis testing workflows here as well, so you and Ian have one place to look for the workflow files, rather than having to ask me where they live.

**Comment by kellijohnson-NOAA** [comment]
Kathryn and I discussed the best location for these workflow files and, because some of them are SS3 focused, we thought that it would be better to house them within "nmfs-stock-synthesis" rather than somewhere like FIT or a less-used organization such as under a personal GitHub account. Ideally, I see the number of workflow files growing in the future and, if they are stored in one central location, it makes it easier to ensure that r4ss, ss3sim, and SSMSE are all up to date with the best practices in a timely manner. We could in theory store all of the R-focused workflows in r4ss and call them from there but we thought this was less visible.

**Comment by Rick-Methot-NOAA** [comment]
Makes sense.  Go for it.

**Comment by k-doering-NOAA** [comment]
Thanks @Rick-Methot-NOAA ! @kellijohnson-NOAA , could you create the repo? Unfortunately I don't have permissions to do it (NWFSC rules)

**Comment by kellijohnson-NOAA** [comment]
No problem, do you want "workflows" or "ss3-workflows"?

**Comment by Rick-Methot-NOAA** [comment]
I just created "workflows"
